decidim-core/lib/decidim/core/api/user_type.rb

```diff
@@ -13,11 +13,15 @@ module Decidim
     field :name, !types.String, "The user's name"
 
     field :avatarUrl, !types.String, "The user's avatar url" do
-      resolve ->(obj, _args, _ctx) { obj.avatar.url }
+      resolve ->(obj, _args, _ctx) { obj.deleted? ? ActionController::Base.helpers.asset_path("decidim/default-avatar.svg") : obj.avatar.url }
     end
 
     field :organizationName, !types.String, "The user's organization name" do
       resolve ->(obj, _args, _ctx) { obj.organization.name }
     end
+
+    field :deleted, !types.Boolean, "Whether the user's account has been deleted or not" do
```

Comment:
`field :deleted, type, "description", property: :deleted?` should work just fine, and you avoid having to use the block